### PR TITLE
Sidebar Upsell Flow: treatment as default

### PR DESCRIPTION
**Pull request by lupus2k**
## Proposed Changes

We move forward with the treatment flow as default for the sidebar upsell.
Context: pb5gDS-2xY-p2#comment-3482

Related to https://github.com/Automattic/dotcom-forge/issues/1709

## To do
- [ ] Update e2e tests for the complete flow.

## Screenshots
| Domains | Plans |
| --- | --- |
| image | image |

## Testing Instructions

* Access the current Domain upsell flow starting at `Upgrade` button on your side menu
* The domain page should look like the `Domains` screenshot
* The plans page should look like the `Plans` screenshot

* You can test the new or old design by adding `&flags=onboarding/2023-pricing-grid` or `&flags=-onboarding/2023-pricing-grid` to the plans URL.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] ~~[Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?~~
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~~
- [ ] ~~Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~~
- [ ] ~~For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?~~

### CODE REVIEW

**Review by allilevine — APPROVED** [review]
Looks good! I tested by adding my user to the control group and then clicking the sidebar upsell. I also tested that the default Plans and Domains pages look the same.

We'll need to follow up on how this looks with the new Plans grid. I'm not sure the Free option should be there.

File: client/state/selectors/is-domain-sidebar-experiment-user.js [hunk 1]
```diff
@@ -1,15 +0,0 @@
...
- * Whether the user is in the Domain sidebar upsell experiment.
- *
- * @param state {object} Global state tree.
- * @returns {boolean} Whether the user is in the Domain sidebar upsell experiment.
- */
-export const isDomainSidebarExperimentUser = ( state ) => {
-	const currentUser = getCurrentUser( state );
-	const domainAndPlanPackage = !! getCurrentQueryArguments( state )?.domainAndPlanPackage;
-
-	return domainAndPlanPackage && 'treatment' === currentUser?.calypso_sidebar_upsell_experiment;
```
Comment (tyxla) [inline client/state/selectors/is-domain-sidebar-experiment-user.js:14]: Seems like we can also remove the `calypso_sidebar_upsell_experiment` key from:

1) The list of allowed additional user attributes:

https://github.com/Automattic/wp-calypso/blob/55db65755421ed2a147412a57dc527abda66be42/client/lib/user/shared-utils/filter-user-object.js#L35

2) The optional user data that is part of the user type:

https://github.com/Automattic/wp-calypso/blob/55db65755421ed2a147412a57dc527abda66be42/client/lib/user/user.d.ts#L47

**Review by tyxla — COMMENTED** [review]
No concerns from the frameworks team perspective 👍 

There's just one little additional removal suggestion.

Also, the branch seems to be needing a rebase.

File: client/state/selectors/is-domain-sidebar-experiment-user.js (hunk 1 above)
Comment (lupus2k) [reply client/state/selectors/is-domain-sidebar-experiment-user.js:14]: Thanks for your suggestion, I totally missed those 😞

**Comment by candy02058912** [comment]
> Update e2e tests for the complete flow.

@lupus2k I've created a follow-up issue (https://github.com/Automattic/wp-calypso/issues/73707) to track it since it was mentioned in the TODO.

**Review by candy02058912 — APPROVED** [review]
Thanks for working on this @lupus2k! LGTM

I removed the styling used for `PlansFeaturesMainWithComparison` because we now have the new plans grid.

I've tested it on my end, and it's working, but I would definitely love a double-check before this PR gets merged.

**Pull request merged**